### ISSUE TITLE
Error when running sdm_neva_merge_biomass_ind.R

### ISSUE THREAD

**Issue by AlexDikelsky** [aallyn/COCA_SDM_MS#1]
I unzipped all of the files in the `Data` directory, but after installing all needed R libraries (I also needed to install `gdal-devel`, `proj`, `proj-devel`, `geos-devel`, `sqlite-devel`, `udunits2-devel`, and  `libjpeg-devel` on Fedora to have the R libraries load correctly), I ran into an error when running

    R --no-save < sdm_neva_merge_biomass_ind.R

The output to the screen stopped with

```R
> ## Loading in the data and some prep -- unzip any zipped folders and add just inner files to the Data folder. Zipping was done to reduce file size for hosting on GitHub.
> # Fish assessment species
> fish.spp<- read.csv(here::here("Data", "Assesmentfishspecies.csv"))
> 
> # Read it in, do some quick formatting to fit the GAM
> dat<- readRDS(here::here("Data", "model.dat.rds"))%>% 
+   filter(., SVSPP %in% fish.spp$SVSPP) %>%
+   left_join(., fish.spp, by = "SVSPP") 
zsh: terminated  R --no-save < sdm_neva_merge_biomass_ind.R
```

I can share the full output, but do you know what could be causing the `zsh: terminated ...` at the end? I tried this with `bash` and `sh`, and those had the same output except instead of `zsh: terminated ...`, they just said

    Terminated

I am running this through a Fedora virtual machine on the command line, and have not yet tried running the code with RStudio. This is the first time `left_join` was used in the file, which may be the problem, but if the function wasn't found, I think a more verbose error would have been raised. Do you know what might be causing this?

## EDIT: It worked fine after I increased the VM's RAM to 16 gigabytes. It also worked on my Windows machine.

**Comment by aallyn** [comment]
Hey Alex, I just saw the edit. Is all okay now and it was just a memory issue?

**Comment by AlexDikelsky** [comment]
Yes, the data looked like it was loading correctly now. There was still an error when I run the code though:
```R
Error in eval(predvars, data, env) : 
  object 'SEASONALMU.OISST.Scale' not found
Calls: sdm_neva_bayes_bio ... eval -> model.frame -> model.frame.default -> eval -> eval
```
I think this has to do with me not having the OISST data. Can you email me the netCDF file you mentioned in the README?

**Comment by aallyn** [comment]
Is this still being run through Fedora on VM? I am pretty sure the error is occurring because of an environment issue with the eval call, which I think defaults to looking to the parent frame when calling the function. Admittedly, getting to the outer reaches of my understanding here. When I have had this issue before I have had to define the problem in the code explicitly. For this case, though, I'm just not sure what the best way to do that would be because the model fitting function is being mapped to each season-species, using that specific season-species "SEASONALMU.OISST.Scale" variable as a predictor variable in the model. So, not entirely clear how you could effectively define that object for each season-species given the current structure of the rest of the code. [This](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/eval#:~:text=eval%20evaluates%20the%20expr%20argument,call%20to%20eval%20was%20made)) and [this](http://adv-r.had.co.nz/Environments.html) might be helpful -- easiest solution I think would be running everything interactively with R/RStudio.

**Comment by AlexDikelsky** [comment]
Thanks! I'm using RStudio through Fedora now. Should I be setting my working directory to the base of the repo or the Code directory?

**Comment by aallyn** [comment]
If you've got the `here` library installed, I think you'll be okay without setting a working directory as any data is read in with `here::here("data", "datafile")`, so it should sort that out for you and start at the base of the repo. Or that's what hopefully happens anyways!

**Comment by AlexDikelsky** [comment]
Thanks! I was able to get some plots for different fish by changing the bounds on [this](https://github.com/aallyn/COCA_SDM_MS/blob/be54d8a6242b59f98604f26cd817dd01844d9706/Code/sdm_neva_merge_biomass_ind.R#L779) `for` loop.

Does the `MU` in `SEASONALMU.OISST.Scale` mean anything?

**Comment by aallyn** [comment]
That's great!

Yea, the MU is just signaling that its the seasonal average (mean, mu) temperature being used as a variable in the model to describe species occurrence patterns -- rather than the seasonal min or max or something.
